src/parser/expression.js

```diff
@@ -896,6 +896,14 @@ pp.parseObj = function (isPattern, refShorthandDefaultPos) {
   node.properties = [];
   this.next();
 
+  // `for` keyword begins an object comprehension.
+  if (this.hasPlugin("lightscript") && this.match(tt._for)) {
+    // ...however, `{ for: x }` is a legal JS object.
```

Comment:
I don't have hard numbers on lookahead perf, but I'm pretty sure it's just about the slowest thing that we can do in the parser 😉 compiler perf can really make a difference to UX. 

hmm...